ssl/ssl_rsa.c

```diff
@@ -68,10 +68,6 @@ int SSL_use_certificate_file(SSL *ssl, const char *file, int type)
         goto end;
     }
 
-    if (type != SSL_FILETYPE_ASN1 && type != SSL_FILETYPE_PEM) {
-        ERR_raise(ERR_LIB_SSL, SSL_R_BAD_SSL_FILETYPE);
-        goto end;
-    }
```

Comment:
Are you sure this can be dropped, too?
In the discussion, you explained that the check further below can be removed because the same check is done already in lines 71-74.
But now you propose dropping this as well.

Reply:
But than in discussion https://github.com/openssl/openssl/pull/21394#discussion_r1258404059 we decided that check
 if (type != SSL_FILETYPE_ASN1 && type != SSL_FILETYPE_PEM) {
        ERR_raise(ERR_LIB_SSL, SSL_R_BAD_SSL_FILETYPE);
        goto end;
}  
should be removed, and check  
else {
        ERR_raise(ERR_LIB_SSL, SSL_R_BAD_SSL_FILETYPE);
        goto end;
}
should be left, because it is more optimal and will help to avoid possible errors in the future

Reply:
Thanks for pointing this out.